addons/auth_signup/controllers/main.py

```diff
@@ -50,11 +50,16 @@ def web_auth_signup(self, *args, **kw):
             raise werkzeug.exceptions.NotFound()
 
         if 'error' not in qcontext and request.httprequest.method == 'POST':
-            try:
-                self.do_signup(qcontext)
-                return super(AuthSignupHome, self).web_login(*args, **kw)
-            except (SignupError, AssertionError), e:
-                qcontext['error'] = _(e.message)
+            if request.env["res.users"].sudo().search(
+                    [("login", "=", qcontext.get("login"))]):
+                qcontext["error"] = _("Another user is already registered "
+                                      "using this email address.")
+            else:
+                try:
+                    self.do_signup(qcontext)
+                    return super(AuthSignupHome, self).web_login(*args, **kw)
+                except (SignupError, AssertionError) as e:
+                    qcontext['error'] = _(e.message)
```

Comment:
actually I don't think we should ever display the exception message to the user, way too technical. We can log it server side and display a more generic "could not register user" or something (except if unicity on login where we display a better error message).
As the list of checks may grow over time (adding other constraints), I prefer to make all the checks in the except block, [easier to ask forgiveness](https://docs.python.org/2/glossary.html#term-eafp)